internal/mode/static/nginx/config/generator_test.go

```diff
@@ -63,7 +64,7 @@ func TestGenerate(t *testing.T) {
 		StreamUpstreams: []dataplane.Upstream{
 			{
 				Name:      "stream_up",
-				Endpoints: nil,
+				Endpoints: make([]resolver.Endpoint, 1),
```

Comment:
I think I check the length of the endpoints when building stream upstreams

Reply:
But you should still be able to check the length of a nil slice, and endpoints can be nil, right?

Reply:
well yeah, for the purposes of the test I set the length to 1 because I think I am looping through them when creating stream server and check if there exists an upstream. with at least one endpoint. I'm not sure if that's correct tho,

Reply:
I'm still struggling to understand why this is needed for the test. What happens if endpoints is nil here?

Reply:
ok if endpoints is nil then the length of it will be 0. Basically when I create stream server blocks, I check to make sure it references a valid upstream. To check it references a valid upstream I create a map of upstream names, and **I only add upstream to the map if the length of the resolver endpoints is not 0**. This was recently added I think because it solved a bug when the service was missing endpoints but still ended up being completely valid.